src/charm.py

```diff
@@ -466,6 +466,9 @@ def _update_tempo_cluster(self):
             logger.error("skipped tempo cluster update: inconsistent state")
             return
 
+        if not self.unit.is_leader():
+            return
```

Comment:
Since you have a leader guard, do we need to make sure that the code below is run on `leader-elected`?

Reply:
Yes. It's already run on `leader-elected`